### PR TITLE
Fixes #8289 - complete handling for cloudinit in ovirt feature

### PR DESCRIPTION
**Pull request by karmab**
found a  way to move all the logic into ovirt compute resource code (by means of using the comment field of the vm to temporarily hold the rendered user_data)
opening PR 
and another small depending one at rbovirt to expose comment field ( description was available but only 255 chars long )

### CODE REVIEW

**Comment by karmab** [comment]
associated rbovirt PR https://github.com/abenari/rbovirt/pull/80

**Comment by dLobatog** [comment]
[test]

**Comment by karmab** [comment]
[test]

**Comment by karmab** [comment]
[test]

**Comment by karmab** [comment]
rbovirt PR got merged

File: bundler.d/ovirt.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 group :ovirt do
-  gem 'rbovirt', '~> 0.0.32'
+  gem 'rbovirt', '~> 0.0.35'
```
Comment (domcleal) [inline bundler.d/ovirt.rb]: This will be .36 once the necessary rbovirt is out.

**Comment by karmab** [comment]
[test]

**Comment by dLobatog** [comment]
I think the gem still needs to be released, from Jenkins: 
`13:36:50 Could not find gem 'rbovirt (~> 0.0.36) ruby' in any of the gem sources listed`

**Comment by karmab** [comment]
right, i m waiting for jhernandez to release it , i ll let you know once he s done

**Comment by karmab** [comment]
note rbovirt gem will have version 0.1.0 ( and not 0.0.36 as originally expected). edited the bundler file accordingly

**Comment by karmab** [comment]
rbovirt 0.1.0 released

**Comment by karmab** [comment]
[test]

**Comment by domcleal** [comment]
We're not compatible with 0.1.0 due to its use of rest-client 1.7+, we'll need to figure out how to update that.  It's a bit of a pain as we need 1.6 for Hammer due to Ruby 1.8.

**Comment by karmab** [comment]
is it crazy to use  httparty instead of rest_client to quickly solve the issue?
if yes, i can work with rbovirt to make that happen

**Comment by domcleal** [comment]
Er, possibly, seems a lot of effort to go to.  I'm hoping we can drop the Ruby 1.8 support in Hammer soon (http://projects.theforeman.org/issues/8978) and then it'll make it possible to update rest-client.

re httparty, we'd have a little trouble packaging the latest httparty versions as RPMs due to the json version required.

**Comment by karmab** [comment]
ok, we ll  wait for this rest-client update then

**Comment by dLobatog** [comment]
I'll set this to Reached an impasse until we come up with a way of integrating rest_client 1.7.

**Comment by karmab** [comment]
as all of this could take too long, we will open a new rbovirt branch to maintain compatibility and have this feature merged

**Comment by dLobatog** [comment]
@karmab I'm not sure I follow?

**Comment by karmab** [comment]
@eLobato what i meant is that we have opened a parralel branch rbovirt 0.0.36 to maintain compatibility with rest_client <1.7.
Once latest changes are pushed into this branch, we can maybe merge this PR specifying this branch while the rest_client issue is fixed

**Comment by dLobatog** [comment]
On 05/29, Karim Boumedhel wrote:

> @eLobato what i meant is that we have opened a parralel branch rbovirt 0.0.36 to maintain compatibility with rest_client <1.7.
> Once latest changes are pushed into this branch, we can maybe merge this PR specifying this branch while the rest_client issue is fixed

If a 0.0.37 gem comes out that would work, remember we need to package it to
serve it in a production installation, so just the GitHub branch would
not suffice.

> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/theforeman/foreman/pull/2350#issuecomment-106730324

## 

Daniel Lobato Garcia

@eLobatoss
blog.daniellobato.me
daniellobato.me

GPG: http://keys.gnupg.net/pks/lookup?op=get&search=0x7A92D6DD38D6DE30
Keybase: https://keybase.io/elobato

**Comment by karmab** [comment]
added https://github.com/fog/fog/pull/3601 which was missing

**Comment by karmab** [comment]
i see the blocking issue from ruby was resolved, can we proceed with the rest-client part?

**Comment by domcleal** [comment]
No, we still have a dependency on rest-client for Ruby 1.8 support in our installer, we can't update it currently.

**Comment by karmab** [comment]
do we still have this dependency on the installer?

**Comment by domcleal** [comment]
Yes, nobody's reimplemented it.

**Comment by mmoll** [comment]
Something like Apipie/apipie-bindings@da7ea84 could be possible

**Comment by mmoll** [comment]
@domcleal correct me if I'm wrong, but I think when http://projects.theforeman.org/issues/11897 would be fixed to support newer rest-client versions (while still retaining 1.6.x compatibility to support Ruby 1.8), rest-client 1.8.0 could be used in the SCL for Foreman core as everything else using Ruby 1.8 is using apipie and not rest-client directly.

As rest-client 1.7 enabled SSL verification by default, there's maybe a bit more to do than just update the gem but rest-client 1.6 is not getting any updates anymore and has two known vulnerabilities (CVE-2015-1820, CVE-2015-3448), so it would be nice to get this all sorted out for the 1.11 release.

**Comment by karmab** [comment]
should i rebase my code against latest foreman and have some hope this
feature could make it for 1.11?

**Comment by mmoll** [comment]
I'd rather wait until a suitable rest-client version is in Foreman core.

**Comment by domcleal** [comment]
> @domcleal correct me if I'm wrong, but I think when http://projects.theforeman.org/issues/11897 would be fixed to support newer rest-client versions (while still retaining 1.6.x compatibility to support Ruby 1.8), rest-client 1.8.0 could be used in the SCL for Foreman core as everything else using Ruby 1.8 is using apipie and not rest-client directly.

A non-SCL 1.6.x version would still be required, and we don't have a way to package two different versions I don't think.

**Comment by mmoll** [comment]
I see :(

**Comment by domcleal** [comment]
I've figured it out, we can package two different versions (https://github.com/domcleal/foreman-packaging/commit/0c789ec68af878b9088a164dba4fe74db640ecc8).

To get this back on track then, the rest-client version needs to be updated (pinned to 1.6.x in Gemfile now) for compatibility with rbovirt 0.1.x.  Are you comfortable doing that @karmab?  Either in a separate ticket/PR or here.

[test]

**Comment by karmab** [comment]
will do !

**Comment by mmoll** [comment]
now that theforeman/smart_proxy_discovery#10 got merged... what's the status here?

**Comment by karmab** [comment]
my understanding is that rebasing should be enough now. right @domcleal  ?

**Comment by domcleal** [comment]
> the rest-client version needs to be updated (pinned to 1.6.x in Gemfile now) for compatibility with rbovirt 0.1.x

**Comment by mmoll** [comment]
@karmab ping?

**Comment by karmab** [comment]
Will do this week. Promise :)

**Comment by mmoll** [comment]
turns out it's a bit more complicated and Katello is involved, see http://projects.theforeman.org/issues/13698

**Comment by domcleal** [comment]
This can be rebased now - Foreman's moved to 0.1.x with the updated rest-client.  I won't be able to review this though, will leave that to somebody else.

File: app/models/compute_resources/foreman/model/ovirt.rb [hunk 2]
```diff
@@ -21,11 +21,14 @@ def self.model_name
...
 
+    def user_data_supported?
+      true
+    end
+
     def host_compute_attrs(host)
       super.tap do |attrs|
         attrs[:os] = { :type => determine_os_type(host) } if supports_operating_systems?
       end
-    end
```
Comment (mmoll) [inline app/models/compute_resources/foreman/model/ovirt.rb:28]: I guess this needs to stay

File: app/models/compute_resources/foreman/model/ovirt.rb [hunk 3]
```diff
@@ -187,10 +190,24 @@ def storage_domains(opts = {})
...
+        vm.start(:blocking => true)
+      end
+    end
+
+    def start_with_cloudinit(uuid, user_data = nil)
+      find_vm_by_uuid(uuid).start_with_cloudinit(:blocking => true, :user_data => user_data)
     end
 
     def create_vm(args = {})
+      #ovirt doesn't accept '.' in vm name.
```
Comment (mmoll) [inline app/models/compute_resources/foreman/model/ovirt.rb]: this was changed in current develop, iirc

File: app/models/compute_resources/foreman/model/ovirt.rb [hunk 4]
```diff
@@ -187,10 +191,22 @@ def storage_domains(opts = {})
     end
 
     def start_vm(uuid)
-      find_vm_by_uuid(uuid).start(:blocking => true)
+      vm = find_vm_by_uuid(uuid)
+      if vm.comment.include? "cloud-config"
+        vm.start_with_cloudinit(:blocking => true, :user_data => vm.comment )
```
Comment (mmoll) [inline app/models/compute_resources/foreman/model/ovirt.rb]: one space too much before the bracket

**Comment by mmoll** [comment]
One code style note. Also, please squash the commits together, we're getting there :)

**Comment by karmab** [comment]
anything more needed to make this happen :) ?

**Comment by dLobatog** [comment]
Merged as 5dec3a526879e70ef0c3aeac0c8d0945d1614dba, thanks @karmab!

**Comment by ohadlevy** [comment]
wow - this one took a while :) thanks for your patient @karmab

**Pull request closed without merging**